capability/policy/selector/selector.go

```diff
@@ -41,6 +56,38 @@ type segment struct {
 	index    int
 }
 
+// NewFieldSegment creates a new segment for a field.
+func NewFieldSegment(field string) segment {
```

Comment:
@MichaelMure we don't use all those constructors atm. But I needed to deal w/ `segment` from the parent pkg (tests) and didn't want to make that type public. wdyt